core/src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -125,15 +126,19 @@ public DateFieldMapper build(BuilderContext context) {
         }
 
         protected void setupFieldType(BuilderContext context) {
-            FormatDateTimeFormatter dateTimeFormatter = fieldType().dateTimeFormatter;
-            // TODO MOVE ME OUTSIDE OF THIS SPACE?
             if (Version.indexCreated(context.indexSettings()).before(Version.V_2_0_0)) {
-                boolean includesEpochFormatter = dateTimeFormatter.format().contains("epoch_");
+                if (fieldType().dateTimeFormatter().equals(Defaults.DATE_TIME_FORMATTER)) {
```

Comment:
Doesn't this have the same problem? If the user upgrades to 2.0, and tries to set their older index to use the new defaults, they will be forced back to non strict?